crates/common/account_manager/src/keystore.rs

```diff
@@ -0,0 +1,327 @@
+    },
+    /// Short parameter names
+    Short {
+        m: u32,
+        t: u32,
+        p: u32,
+        salt: String,
+    },
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CipherParams {
+    /// Cipher function name, must be "aes-256-gcm"
+    pub function: String,
+
+    /// Cipher parameters
+    pub params: CipherParamsInner,
+
+    /// Encrypted data as hex string
+    pub ciphertext: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CipherParamsInner {
+    /// Nonce/IV as hex string
+    pub nonce: String,
+
+    /// Authentication tag as hex string
+    pub tag: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct KeyType {
+    /// Key type function name
+    pub function: String,
+
+    /// Key type parameters
+    pub params: KeyTypeParams,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct KeyTypeParams {
+    /// Key lifetime
+    pub lifetime: u32,
+
+    /// Activation epoch
+    pub activation_epoch: u32,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct KeystoreMeta {
+    /// Creation timestamp
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub created: Option<DateTime<Utc>>,
+}
+
+impl QsKeystore {
+    /// Create a new quantum-secure keystore
+    pub fn new(crypto: CryptoParams, keytype: KeyType, uuid: Uuid) -> Self {
+        Self {
+            version: 5,
+            crypto,
+            keytype,
+            description: None,
+            quantum_secure: true,
+            uuid,
+            path: None,
+            meta: Some(KeystoreMeta {
+                created: Some(Utc::now()),
+            }),
+        }
+    }
+
+    /// Create a new keystore from seed phrase and key parameters
+    pub fn from_seed_phrase(
+        seed_phrase: &str,
+        lifetime: u32,
+        activation_epoch: u32,
+        description: Option<String>,
+        path: Option<String>,
+    ) -> Self {
+        let uuid = Uuid::new_v4();
+
+        // Generate random salt for KDF (32 bytes)
+        let salt = hex::encode(rand::random::<[u8; 32]>());
+
+        // Generate random nonce for AES-GCM (12 bytes)
+        let nonce = hex::encode(rand::random::<[u8; 12]>());
+
+        // Generate random tag for AES-GCM (16 bytes)
+        let tag = hex::encode(rand::random::<[u8; 16]>());
+
+        // Store the seed phrase as encrypted data (hex encoded)
+        let ciphertext = hex::encode(seed_phrase.as_bytes());
+
+        let kdf = KdfParams::new_full(65536, 3, 4, salt);
+        let cipher = CipherParams::new(nonce, tag, ciphertext);
+        let crypto = CryptoParams { kdf, cipher };
+        let keytype = KeyType::new(lifetime, activation_epoch);
+
+        let mut keystore = Self::new(crypto, keytype, uuid);
+        keystore.description = description;
+        keystore.path = path;
+
+        keystore
+    }
+
+    /// Validate the keystore structure
+    pub fn validate(&self) -> Result<(), String> {
```

Comment:
we should be using anyhow, our project should only use this_error or anyhow depending on the context